File: ArraysAndStrings/TwoSum.java
import java.util.HashMap;

// Link: https://leetcode.com/problems/two-sum/
// Difficulty: Easy
// Time complexity: O(n)
// Space complexity: O(n)

class TwoSum {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> numVals = new HashMap<Integer, Integer>();
        for (int i = 0; i < nums.length; i++) {
            numVals.put(nums[i], i);
        }
        for (int j = 0; j < nums.length; j++) {
            int complement = target - nums[j];
            if (numVals.containsKey(complement) && numVals.get(complement) != j) {
                return new int[] {j, numVals.get(complement)};
            }
        }
        return null;
    }
}

File: ArraysAndString/TwoSum.py
# Link: https://leetcode.com/problems/two-sum/
# Difficulty: Easy
# Time complexity: O(n)
# Space complexity: O(n)
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        hashmap = {}
        for i in range (len(nums)):
            complement = target - nums[i]
            if complement in hashmap:
                return [i, hashmap[complement]]
            hashmap[nums[i]] = i
        